Panic on unsafe type cast

### Expected behavior

Ship can init a local directory via go-getter

### Actual behavior

Ship panics on a local directory (and likely in other places as well)

```
panic: interface conversion: state.State is state.VersionedState, not state.Empty

goroutine 1 [running]:
github.com/replicatedhq/ship/pkg/ship.(*Ship).Init(0xc00032ad80, 0x2b7f340, 0xc0009604c0, 0x0, 0x0)
        /Users/dex/go/src/github.com/replicatedhq/ship/pkg/ship/kustomize.go:52 +0x812
github.com/replicatedhq/ship/pkg/ship.(*Ship).InitAndMaybeExit(0xc00032ad80, 0x2b7f380, 0xc0000cc008, 0x0, 0xc0004c1c90)
        /Users/dex/go/src/github.com/replicatedhq/ship/pkg/ship/kustomize.go:20 +0x43
github.com/replicatedhq/ship/pkg/cli.Init.func1(0xc0004f7d40, 0xc0004c1ac0, 0x1, 0x1, 0x0, 0x0)
        /Users/dex/go/src/github.com/replicatedhq/ship/pkg/cli/init.go:40 +0x84
github.com/replicatedhq/ship/vendor/github.com/spf13/cobra.(*Command).execute(0xc0004f7d40, 0xc0004c1a80, 0x1, 0x1, 0xc0004f7d40, 0xc0004c1a80)
        /Users/dex/go/src/github.com/replicatedhq/ship/vendor/github.com/spf13/cobra/command.go:698 +0x47a
github.com/replicatedhq/ship/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0004f7440, 0x1, 0x28c6c80, 0x1)
        /Users/dex/go/src/github.com/replicatedhq/ship/vendor/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/replicatedhq/ship/vendor/github.com/spf13/cobra.(*Command).Execute(0xc0004f7440, 0x1007580, 0xc0000b0058)
        /Users/dex/go/src/github.com/replicatedhq/ship/vendor/github.com/spf13/cobra/command.go:736 +0x2b
github.com/replicatedhq/ship/pkg/cli.Execute()
        /Users/dex/go/src/github.com/replicatedhq/ship/pkg/cli/root.go:84 +0x27
main.main()
        /Users/dex/go/src/github.com/replicatedhq/ship/cmd/ship/main.go:6 +0x20
```

### Information

- Ship version: bda5bb61
- Command line run:

```
ship init ./fixtures/terraform
```

### Steps to reproduce the behavior

```
bin/ship init ./fixtures/terraform
```

--- a/fixtures/terraform/ship.yaml
+++ b/fixtures/terraform/ship.yaml
@@ -72,6 +72,6 @@ lifecycle:
        contents: "hi"
     - config: {}
     - render: {}
-#    - terraform: {}
+    - terraform: {}
     - message:
        contents: "hi"

--- a/pkg/lifecycle/render/docker/step_test.go
+++ b/pkg/lifecycle/render/docker/step_test.go
@@ -32,19 +32,19 @@ func TestDockerStep(t *testing.T) {
 		Expect                  error
 	}{
 		{
-			name: "registry succeeds",
+			name:                    "registry succeeds",
 			RegistrySecretSaveError: nil,
 			InstallationIDSaveError: nil,
 			Expect:                  nil,
 		},
 		{
-			name: "registry fails, install id succeeds",
+			name:                    "registry fails, install id succeeds",
 			RegistrySecretSaveError: errors.New("noooope"),
 			InstallationIDSaveError: nil,
 			Expect:                  nil,
 		},
 		{
-			name: "registry fails, install id fails",
+			name:                    "registry fails, install id fails",
 			RegistrySecretSaveError: errors.New("noooope"),
 			InstallationIDSaveError: errors.New("nope nope nope"),
 			Expect:                  errors.New("docker save image, both auth methods failed: nope nope nope"),

--- a/pkg/ship/kustomize.go
+++ b/pkg/ship/kustomize.go
@@ -50,7 +50,7 @@ func (s *Ship) Init(ctx context.Context) error {
 	existingState, _ := s.State.TryLoad()
 	if existingState != nil {
 		if !existingState.IsEmpty() {
-			s.UI.Output(fmt.Sprintf("exsiting state = %#v\n", existingState))
+			s.UI.Output(fmt.Sprintf("existing state = %#v\n", existingState))
 			debug.Log("event", "existing.state")
 
 			if s.Viper.GetString("state-from") != "file" {

--- a/pkg/ship/kustomize.go
+++ b/pkg/ship/kustomize.go
@@ -49,8 +49,8 @@ func (s *Ship) Init(ctx context.Context) error {
 
 	existingState, _ := s.State.TryLoad()
 	if existingState != nil {
-		if existingState != existingState.(state.Empty) {
-			fmt.Printf("exsiting state = %#v\n", existingState)
+		if !existingState.IsEmpty() {
+			s.UI.Output(fmt.Sprintf("exsiting state = %#v\n", existingState))
 			debug.Log("event", "existing.state")
 
 			if s.Viper.GetString("state-from") != "file" {

--- a/pkg/state/models.go
+++ b/pkg/state/models.go
@@ -16,6 +16,7 @@ type State interface {
 	CurrentHelmValuesDefaults() string
 	Upstream() string
 	Versioned() VersionedState
+	IsEmpty() bool
 }
 
 var _ State = VersionedState{}
@@ -31,6 +32,7 @@ func (Empty) CurrentHelmValues() string             { return "" }
 func (Empty) CurrentHelmValuesDefaults() string     { return "" }
 func (Empty) Upstream() string                      { return "" }
 func (Empty) Versioned() VersionedState             { return VersionedState{V1: &V1{}} }
+func (Empty) IsEmpty() bool                         { return true }
 
 type V0 map[string]interface{}
 
@@ -41,9 +43,14 @@ func (v V0) CurrentHelmValues() string             { return "" }
 func (v V0) CurrentHelmValuesDefaults() string     { return "" }
 func (v V0) Upstream() string                      { return "" }
 func (v V0) Versioned() VersionedState             { return VersionedState{V1: &V1{Config: v}} }
+func (v V0) IsEmpty() bool                         { return false }
 
 type VersionedState struct {
 	V1 *V1 `json:"v1,omitempty" yaml:"v1,omitempty" hcl:"v1,omitempty"`
+}
+
+func (v VersionedState) IsEmpty() bool {
+	return false
 }
 
 type V1 struct {

